modules/services/git-sync.nix

```
@@ -0,0 +1,85 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.git-sync;
+  mkUnit = repo: {
+    Unit = { Description = "Git Sync ${repo.name}"; };
+
+    Install = { WantedBy = [ "default.target" ]; };
+
+    Service = {
+      Environment = [
+        "GIT_SYNC_DIRECTORY=${repo.path}"
+        "GIT_SYNC_COMMAND=${cfg.package}/bin/git-sync"
+        "GIT_SYNC_REPOSITORY=${repo.uri}"
+        "GIT_SYNC_INTERVAL=${toString repo.interval}"
+      ];
+      ExecStart = "${cfg.package}/bin/git-sync-on-inotify";
+      Restart = "on-abort";
+    };
+  };
+  services = listToAttrs (map (repo: {
+    name = "git-sync-${repo.name}";
+    value = mkUnit repo;
+  }) cfg.repositories);
+
+in {
+  meta.maintainers = [ maintainers.imalison ];
+
+  options = {
+    services.git-sync = {
+      enable = mkEnableOption "git-sync services";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.git-sync;
+        defaultText = literalExample "pkgs.git-sync";
+        description = ''
+          Package containing the <command>git-sync</command> program.
+        '';
+      };
+
+      repositories = mkOption {
+        description = "A list of objects describing repositories that should be synced.";
+        type = with types;
+          listOf (submodule {
+            options = {
+              name = mkOption {
```

Since the name has to be unique I would suggest using `attrsOf (submodule …` instead of a list. You can use an internal submodule option to hold the attribute name, or even have an option for the entire service unit definition.

---

Probably best to just have the name option, i.e., doing
```suggestion
              name = mkOption {
                internal = true;
                default = name;
```